### PR TITLE
docs: Add v1.4 release notes

### PR DESCRIPTION
**Pull request by bryanhuhta**

### CODE REVIEW

File: docs/sources/release-notes/v1-4.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Notable changes are listed below. For more details, check out the Full 1.4.0 Changelog: https://github.com/grafana/pyroscope/compare/v1.3.0...v1.4.0.
+
+## Features and enhancements
+
+Features and enhancements for version 1.4 include:
+
+* Adhoc profiles API (#2963)
+* Add query-tracer to profilecli ([#2966](https://github.com/grafana/pyroscope/pull/2966))
+* Add ServiceMonitor support to helm chart ([#2944](https://github.com/grafana/pyroscope/pull/2944), [#2929](https://github.com/grafana/pyroscope/pull/2929), [#2967](https://github.com/grafana/pyroscope/pull/2967))
```
Comment (knylander-grafana) [inline docs/sources/release-notes/v1-4.md:28]:
```suggestion
* Add ServiceMonitor support to Helm chart ([#2944](https://github.com/grafana/pyroscope/pull/2944), [#2929](https://github.com/grafana/pyroscope/pull/2929), [#2967](https://github.com/grafana/pyroscope/pull/2967))
```

File: docs/sources/release-notes/v1-4.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+We are excited to present Grafana Pyroscope 1.4.
+
+This release includes several new features will are precursors to larger projects like:
+
+* Adhoc profiling
+* Time series scoped to a flamegraph function
+* Traces to profiles tooling
+
+Additionally, there has been a multitude of smaller changes to improve stability, performance, and documentation.
```
Comment (knylander-grafana) [inline docs/sources/release-notes/v1-4.md:18]: Multitude sounds odd in this sentence. What do you think about this? 
```suggestion
Additionally, numerous other changes improve stability, performance, and documentation.
```

**Review by knylander-grafana — APPROVED** [review]
Thank you for adding release notes.

File: docs/sources/release-notes/v1-4.md (hunk 2 above)
Comment (bryanhuhta) [reply docs/sources/release-notes/v1-4.md:18]: Your suggestion has better flow and is more concise. I like it 👍

**Pull request merged**